packages/core/src/extension/extension-messenger.ts

```diff
@@ -93,16 +92,16 @@ export class ExtensionMessenger {
         return extension;
     }
 
-    private _validateCommand(command: string): ExtensionCommandType {
+    private _validateCommand<T extends keyof ExtensionCommandMap>(command: string): T {
         switch (command) {
             case 'RELOAD_CHECKOUT':
-                return ExtensionCommandType.ReloadCheckout;
+                return ExtensionCommandType.ReloadCheckout as T;
 
             case 'SHOW_LOADING_INDICATOR':
-                return ExtensionCommandType.ShowLoadingIndicator;
+                return ExtensionCommandType.ShowLoadingIndicator as T;
 
             case 'SET_IFRAME_STYLE':
-                return ExtensionCommandType.SetIframeStyle;
+                return ExtensionCommandType.SetIframeStyle as T;
```

Comment:
`as` is a side effect of `const num`. 😂

Reply:
Generics are hard, especially on our present typescript version 🤦

Reply:
On another side note can we use typeguard to check of the generic is of a particular kind?

Reply:
I didn't think I could but I added a `tyepguard<T>` to replace `as T`.
Please check the update.

Reply:
Nicely done 👏